include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+    }
+
+    /*!\brief Removes specified elements from the container.
+     * \param begin_it Begin of range to erase.
+     * \param   end_it Behind the end of range to erase.
+     * \returns        Iterator following the last element removed. If the iterator pos refers to the last element,
+     *                 the end() iterator is returned.
+     *
+     * Invalidates iterators and references at or after the point of the erase, including the end() iterator.
+     *
+     * The iterator first does not need to be dereferenceable if first==end_it: erasing an empty range is a no-op.
+     *
+     * ### Complexity
+     *
+     * Linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     */
+    iterator erase(const_iterator begin_it, const_iterator end_it)
+    {
+        if (begin_it >= end_it) // [[unlikely]]
+            return begin() + std::distance(cbegin(), end_it);
+
+        auto const begin_it_pos = std::distance(cbegin(), begin_it);
+        auto const end_it_pos = std::distance(cbegin(), end_it);
+
+        data.erase(data.cbegin() + begin_it_pos,
+                   data.cbegin() + end_it_pos);
+
+        return begin() + begin_it_pos;
+    }
+
+    /*!\brief Removes specified elements from the container.
+     * \param   pos Remove the element at pos.
+     * \returns     Iterator following the last element removed. If the iterator pos refers to the last element,
+     *              the end() iterator is returned.
+     *
+     * Invalidates iterators and references at or after the point of the erase, including the end() iterator.
+     *
+     * The iterator pos must be valid and dereferenceable. Thus the end() iterator (which is valid, but is not
+     * dereferencable) cannot be used as a value for pos.
+     *
+     * ### Complexity
+     *
+     * Linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     */
+    iterator erase(const_iterator pos)
+    {
+       return erase(pos, pos + 1);
+    }
+
+    /*!\brief Appends the given element value to the end of the container.
+     * \param value The value to append.
+     *
+     * If the new size() is greater than capacity() then all iterators and references (including the past-the-end
+     * iterator) are invalidated. Otherwise only the past-the-end iterator is invalidated.
+     *
+     * ### Complexity
+     *
+     * Amortised constant, worst-case linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     */
+    void push_back(value_type const value)
+    {
+        data.push_back(to_rank(value));
```

Comment:
move the value?

Reply:
value types are always alphabet_concept which makes them cheap to copy.

Reply:
true, and they will be copied anyway. But just thought what if something else is used (external lib) that is bigger than standards built in types